staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/types.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategicpatch
+
+import (
+	"strings"
+
+	"k8s.io/kubernetes/pkg/kubectl/cmd/util/openapi"
+)
+
+const (
+	patchStrategyOpenapiextensionKey = "x-kubernetes-patch-strategy"
+	patchMergeKeyOpenapiextensionKey = "x-kubernetes-patch-merge-key"
+)
+
+type LookupPatchItem interface {
+	openapi.SchemaVisitor
+
+	Errors() []error
+	Path() *openapi.Path
+}
+
+type baseLookupPatchMetaItem struct {
+	*openapi.BaseItem
+}
+
+func (item *baseLookupPatchMetaItem) AddLookupPatchMetaError(err error) {
+	item.GetErrors().AppendErrors(LookupPatchMetaError{Path: item.GetPath().String(), Err: err})
+}
+
+type patchItem struct {
+	baseLookupPatchMetaItem
+	key       string
+	patchmeta PatchMeta
+	subschema openapi.Schema
+}
+
+func NewPatchItem(key string, path *openapi.Path) patchItem {
```

Comment:
Ahah this method is a little bit unfortunate:

`patchItem` -> `baseLookupPatchMetaItem` -> `BaseItem` :-/